Benqi-Agents/Large-Grant/src/agent.spec.ts

```diff
@@ -0,0 +1,503 @@
+import { BigNumber, BigNumberish } from "ethers";
+import { Interface } from "ethers/lib/utils";
+import { JsonRpcProvider } from "@ethersproject/providers/lib/json-rpc-provider";
+import { FindingType, FindingSeverity, Finding, HandleTransaction } from "forta-agent";
+import { provideHandleTransaction } from "./agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { QI_BALANCE_ABI, QI_GRANTED_ABI, QI_TOTAL_SUPPLY, QI_TRANSFER_ABI, ThresholdMode } from "./utils";
+import { MockEthersProvider } from "forta-agent-tools/lib/tests";
+
+const QI_ADDRESS = createAddress("0x1");
+const COMPTROLLER_ADDRESS = createAddress("0x2");
+const RECIPIENT_ADDRESS = createAddress("0x3");
+const IRRELEVANT_ADDRESS = createAddress("0x4");
+
+const COMPTROLLER_IFACE = new Interface([QI_GRANTED_ABI]);
+const QI_IFACE = new Interface([QI_BALANCE_ABI, QI_TRANSFER_ABI]);
+
+export function createFinding(
+  recipient: string,
+  amount: string,
+  thresholdMode: ThresholdMode,
+  threshold: string
+): Finding {
+  return Finding.fromObject({
+    name: "Large QI Grant",
+    description: "There was a large QI Grant in the BENQI Comptroller contract",
+    alertId: "BENQI-4",
+    protocol: "Benqi Finance",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    metadata: {
+      recipient,
+      amount: amount,
+      mode: ThresholdMode[thresholdMode],
+      threshold,
+    },
+  });
+}
+
+const percentageOf = (value: BigNumberish, percentage: BigNumberish): string => {
+  return BigNumber.from(value).mul(percentage).div(100).toString();
+};
+
+describe("Delegate-Votes-Monitor Agent test suite", () => {
+  describe("absolute threshold", () => {
+    describe("handleTransaction", () => {
```

Comment:
why there is this line with every test suite ?

Reply:
That's because I separated the tests between each mode, and each mode would be testing its `handleTransaction`, then `test suite -> mode -> handler`. But, since we're only testing `handleTransaction`, I think we can remove the `handleTransaction` level. What do you think, @yasmine9 ?

Reply:
Yes I prefer removing the `handleTransaction` level, modes separation is enough

Reply:
Good, I will remove it then.

Reply:
Removed in [374c99d](https://github.com/NethermindEth/Forta-Agents/pull/183/commits/374c99d447ca61bcaeb6b8964ceb3c58e41ef780).